src/lib/y2storage/callbacks/probe.rb

```diff
@@ -19,11 +19,116 @@
 
 require "y2storage/callbacks/libstorage_callback"
 
+Yast.import "Mode"
+Yast.import "Label"
+Yast.import "Popup"
+
 module Y2Storage
   module Callbacks
     # Class to implement callbacks used during libstorage-ng probe
-    class Probe < Storage::ProbeCallbacks
+    class Probe < Storage::ProbeCallbacksV3
       include LibstorageCallback
+
+      # Callback for missing commands during probing.
+      #
+      # @param message [String] error title coaming from libstorage-ng
+      #   (in the ASCII-8BIT encoding! see https://sourceforge.net/p/swig/feature-requests/89/)
+      # @param what [String] details coming from libstorage-ng (in the ASCII-8BIT encoding!)
+      # @param command [String] missing command coming from libstorage-ng (in the ASCII-8BIT encoding!)
+      # @param used_features used features bit field coming from libstorage-ng
+      #
+      # @return [Boolean] true will make libstorage-ng ignore the error, false
+      #   will result in a libstorage-ng exception
+      #
+      def missing_command(message, what, command, used_features)
+        textdomain "storage"
+
+        # force the UTF-8 encoding to avoid Encoding::CompatibilityError exception
+        message.force_encoding("UTF-8")
+        what.force_encoding("UTF-8")
+        command.force_encoding("UTF-8")
+
+        log.info "libstorage-ng reported a missing command, asking the user whether to continue"
+        log.info "Error details. message: #{message}. what: #{what}. command: #{command}. "\
+                 "used_features: #{used_features}."
+
+        # Redirect to error callback if no packages can be installed.
+        return error(message, what) if used_features == 0 || Yast::Mode.installation
+
+        package_handler = Y2Storage::PackageHandler.new
+        package_handler.add_feature_packages(used_features)
+
+        # Redirect to error callback if no packages can be installed.
+        return error(message, what) if package_handler.compact.empty?
+
+        description = _("An external command required for probing is missing. When\n"\
+                        "continuing despite the error, the presented system information\n"\
+                        "will be incomplete. You may also install the required packages\n"\
+                        "and restart probing.")
+
+        what += "\n\n" + missing_command_handle_packages_text(package_handler)
+
+        question = _("Continue despite the error, install required packages or abort?")
+        buttons = { continue: Yast::Label.ContinueButton, install: _("Install Packages"),
+                    abort: Yast::Label.AbortButton }
+
+        result = Yast2::Popup.show(full_message(message, description, question, what),
+          details: wrap_text(what), buttons: buttons, focus: :continue)
```

Comment:
I am not sure about focus here. Why not having install as default option? It should be the ideal path for user, not?

Reply:
I presented a screenshot yesterday and did not get any feedback.

Reply:
I agree with @jreidinger, IMHO *install* should be the default option. Sorry @aschnell if you didn't get feedback earlier. Sometimes we don't realize about some details until we have a deeper revision. Changing the default option should be easy to do, right?